src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.cs

```diff
@@ -245,10 +245,27 @@ internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSynt
                     var assembly = compilation.GetAssemblyOrModuleSymbol(reference) as IAssemblySymbol;
                     if (assembly != null)
                     {
-                        AddRange(allSymbolReferences, await finder.FindInMetadataAsync(otherProject.Solution, assembly, reference, exact).ConfigureAwait(false));
+                        findTasks.Add(finder.FindInMetadataAsync(otherProject.Solution, assembly, reference, exact));
                     }
                 }
+            }
+
+            while (findTasks.Count > 0)
+            {
+                // Keep on looping through the 'find' tasks, processing each when they finish.
+                cancellationToken.ThrowIfCancellationRequested();
+                var doneTask = await Task.WhenAny(findTasks).ConfigureAwait(false);
+
+                // One of the tasks finished.  Remove it from the list we're waiting on.
+                findTasks.Remove(doneTask);
+
+                // Add its results to the final result set we're keeping.
+                AddRange(allSymbolReferences, await doneTask.ConfigureAwait(false));
 
+                // If we've got enough, no need to keep searching. 
+                // TODO(cyrusn): Should we cancel the existing work?  IMO, no.  These tasks will
```

Comment:
Just drop the "TODO(cyrusn)" and change "IMO, no" to "probably not", and then it doesn't sound so TODOish. It's good the comment explains why this was chosen, but I don't think anything needs to be revisited.